templates/app-service-plan.json

```diff
@@ -132,6 +161,29 @@
           "Service Line": "[parameters('serviceLineTag')]",
           "Service Offering": "[parameters('serviceOfferingTag')]"
       }
+    },
+    {
+      "type": "Microsoft.Web/serverfarms/providers/diagnosticSettings",
+      "condition": "[greater(length(parameters('workspaceId')), 0)]",
+      "apiVersion": "2017-05-01-preview",
+      "name": "[concat(parameters('appServicePlanName'),'/microsoft.insights/', parameters('settingName'))]",
+      "dependsOn": [
+        "[parameters('appServicePlanName')]"
+      ],
+      "properties": {
+        "name": "[parameters('settingName')]",
+        "workspaceId": "[parameters('workspaceId')]",
+        "metrics": [
```

Comment:
There's a whole slew of diagnostics logging we can add here - for example, in FDS I also define:

```

                "logs":[
                    {
                        "category": "AppServiceHTTPLogs",
                        "enabled": true,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    },
                    {
                        "category": "AppServiceConsoleLogs",
                        "enabled": true,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    },
                    {
                        "category": "AppServiceAppLogs",
                        "enabled": true,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    },
                    {
                        "category": "AppServiceAuditLogs",
                        "enabled": true,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    },
                    {
                        "category": "AppServiceIPSecAuditLogs",
                        "enabled": true,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    },
                    {
                        "category": "AppServicePlatformLogs",
                        "enabled": true
                        ,
                        "retentionPolicy": {
                            "days": 30,
                            "enabled": true
                        }
                    }
                ],
```